firedrake/slate/preconditioners.py

```diff
@@ -211,12 +212,9 @@ def initialize(self, pc):
         self.S.force_evaluation()
         Smat = self.S.petscmat
 
-        # Nullspace for the multiplier problem
-        nullspace = create_schur_nullspace(P, -K * Atilde,
-                                           V, V_d, TraceSpace,
-                                           pc.comm)
+        nullspace = self.app_cxt.get("hybridization_trace_nullspace", None)
```

Comment:
So the only thing I don't like about this is that the user doesn't get access to the trace function space outside (they never see it).

Why not instead allow the user to provide a callback for making the null space:

```
def nullspace(V):
   return ...

appctx["nullspace"] = ...
```

And then you do:

```
nullspace = self.app_cxt.get("...", None)
if nullspace is not None:
    nullspace = nullspace(TraceSpace)
```